Detectors/EMCAL/calib/include/EMCALCalib/BadChannelMap.h

```diff
@@ -0,0 +1,186 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+#include <TObject.h>
+
+#include <iosfwd>
+#include <unordered_map>
+
+class TH2;
+
+namespace o2
+{
+
+namespace emcal
+{
+
+/// \class BadChannelMap
+/// \brief CCDB container for masked cells in EMCAL
+/// \author Markus Fasel <>, Oak Ridge National Laboratory
+/// \since June 25th, 2019
+///
+/// # The EMCAL Bad Channel Map
+///
+/// The bad channel map contains channels which are found to be
+/// bad or problematic during the calibration process and need
+/// to be masked so they are not considered in the clusterization
+/// or other analysis processes. Cells can be either bad, meaning
+/// that the spectra shape is so much distorted that recovery is not
+/// possible, or warm, indicating that the cell might be recovered
+/// by recalibration of the energy. 
+///
+/// Bad channels can be added via
+/// ~~~.{cxx}
+/// o2::emcal::BadChannelMap bcm;
+/// bcm.addBadChannel(1234, o2::emcal::BadChannelMap::MaskStatus_t::BAD_CELL);
+/// ~~~
+/// The cell container stores only bad or warm cells, good cells 
+/// are ignored. However the function addBadChannel also updates 
+/// the container with a new channel status - calling addBadChannel
+///  with the status GOOD_CELL for a cell which is stored in the container 
+/// the cell is removed.
+///
+/// Reading the channel status is done via
+/// ~~~.{cxx}
+/// auto status = bcm.getChannelStatus(1234); // returning BAD_CELL in this case
+/// ~~~
+/// Calling getChannelStatus for cells not registered in the 
+/// container will return the status GOOD_CELL.
+///
+/// The BadChannelMap is Browsable, when calling in the ROOT 
+/// browser a 2D histogram with the bad/warm cells with value 
+/// 1 / 2 shown depending on column/row. As the histogram is 
+/// created dynamically from the absolute cell ID an instance 
+/// of the EMCAL Geometry is required - otherwise an empty
+/// histogram is created.
+///
+/// The bad channel map can be created from multiple bad channel
+/// maps using the operator +=. This allows for the combination
+/// of the bad channel map from multiple time slices. For cells
+/// present in both bad channel maps always the worst case condition
+/// (BAD_CELL) is assumed.
+class BadChannelMap : public TObject
+{
+ public:
+  enum class MaskType_t {
```

Comment:
Since you have just 3 values in the enum, why not to declare it as ``enum class MaskType_t : char {...`` to have the map more compact?

Reply:
Thanks a lot! Will fix this!